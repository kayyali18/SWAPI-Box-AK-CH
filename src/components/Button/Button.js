import React, { Component } from 'react';

class Button extends Component {
  constructor(props) {
    super(props) 

    this.state = {
      isSelected: false
    }
  }

  handleClick = () => {
    this.setState({
      isSelected: !this.state.isSelected
    })
  }

  render() {
    return (
      <div>
        <button onClick={() => {this.handleClick()}}>{this.props.buttonText}</button>
      }
      </div>
      )
  }
}

export default Button;