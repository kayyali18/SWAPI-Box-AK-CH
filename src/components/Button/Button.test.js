import React from "react"
import ReactDOM from "react-dom"
import Button from "./Button"
import { shallow } from "enzyme"

describe("App", () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallow(<Button />)
    
  })

  it("renders without crashing", () => {
    expect(wrapper).toMatchSnapshot();
    
  })

  it("should reset state after a click", async() => {
    await wrapper.instance().handleClick()

    expect(wrapper).toMatchSnapshot();
  })

  it('should call an anonymous function on click', () => {
    wrapper.find('button').prop('onClick')()
  })
})
