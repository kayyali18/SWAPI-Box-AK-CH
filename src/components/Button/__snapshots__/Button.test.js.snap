// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders without crashing 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    People
  </button>
</div>
`;

exports[`App should reset state after a click 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    People
  </button>
</div>
`;
